server/src/testFixtures/java/io/crate/testing/LogicalPlanAssert.java

```diff
@@ -24,22 +24,36 @@
 import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
 
 import org.assertj.core.api.AbstractAssert;
+import org.jetbrains.annotations.Nullable;
 
 import io.crate.planner.operators.LogicalPlan;
 import io.crate.planner.operators.PrintContext;
+import io.crate.planner.optimizer.costs.PlanStats;
 
 public class LogicalPlanAssert extends AbstractAssert<LogicalPlanAssert, LogicalPlan> {
 
+    @Nullable
+    private final PlanStats planStats;
+
     protected LogicalPlanAssert(LogicalPlan actual) {
+        this(actual, null);
+    }
+
+    protected LogicalPlanAssert(LogicalPlan actual, @Nullable PlanStats planStats) {
         super(actual, LogicalPlanAssert.class);
+        this.planStats = planStats;
     }
 
-    private static String printPlan(LogicalPlan logicalPlan) {
-        var printContext = new PrintContext(null);
+    private String printPlan(LogicalPlan logicalPlan) {
+        var printContext = new PrintContext(planStats);
         logicalPlan.print(printContext);
         return printContext.toString();
     }
 
+    public LogicalPlanAssert withPlanStats(PlanStats planStats) {
+        return new LogicalPlanAssert(this.actual, planStats);
```

Comment:
it's fine as is, just wondering if instead of having the `planStats` final and therefore a ctor argument, we can make this a simple setter, but as you prefer.

Reply:
I think we prefer having fields final if possible in the project.

Reply:
totally fine as is!